src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,213 @@
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+
+    @Override
+    public Car create(Car car) {
+        String insertRequest = "INSERT INTO cars (manufacturer_id, model) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement
+                        = connection
+                        .prepareStatement(insertRequest, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setLong(1, car.getManufacturer().getId());
+            createCarStatement.setString(2, car.getModel());
+            createCarStatement.executeUpdate();
+            ResultSet generatedKeys = createCarStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                car.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create car in db" + car, e);
+        }
+        insertCar(car);
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getRequest = "SELECT * FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id"
+                + " WHERE c.id = ? AND c.is_deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement
+                        = connection.prepareStatement(getRequest)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get car with id = " + id, e);
+        }
+        if (car != null) {
+            car.setDriverList(getDriversByCarId(car.getId()));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllRequest = "SELECT * FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection
+                        .prepareStatement(getAllRequest)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can`t get all from db", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can`t get all from db", e);
```
lets use custom exception